src/run/run_step.go

```diff
@@ -46,7 +54,20 @@ func Parallel(ctx context.Context, state *core.BuildState, labels []core.Annotat
 		g.Go(func() error {
 			limiter <- struct{}{}
 			defer func() { <-limiter }()
-			return run(state, label, args, true, quiet, remote, env, detach, inTmp, dir)
+
+			quiet := output == Quiet || output == GroupImmediate
+			out, combined, err := run(state, label, args, true, quiet, remote, env, detach, inTmp, dir)
+
+			if output == GroupImmediate {
+				fmt.Println(label)
+				if err != nil {
+					os.Stderr.Write(combined)
+				} else {
+					os.Stdout.Write(out)
```

Comment:
Doesn't this print it twice? I think out is already encoded in `err` below by `toExitError()`.

Reply:
It now logs all errors without duplication. Did some refactoring for clarity and easy future enhancements.